src/uct/sm/base/sm_iface.c

```diff
@@ -54,66 +56,28 @@ uct_sm_base_query_tl_devices(uct_md_h md, uct_tl_device_resource_t **tl_devices_
 /* read boot_id GUID or use machine_guid */
 static uint64_t uct_sm_iface_get_system_id()
 {
-    static ucs_init_once_t init_once = UCS_INIT_ONCE_INITIALIZER;
-    static uint64_t system_id        = 0;
-    const char *boot_id_filename     = "/proc/sys/kernel/random/boot_id";
-    const char *guid_scan_fmt        = "%x-%4hx-%4hx-%4hx-%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx";
-    char guid_str[256];
-    ssize_t size;
-    int res;
-
-    union {
-        struct {
-            uint32_t v1;
-            uint16_t v2;
-            uint16_t v3;
-            uint16_t v4;
-            uint8_t  v5[6];
-        };
-        struct {
-            uint64_t low_dword;
-            uint64_t hi_dword;
-        };
-    } UCS_S_PACKED guid;
-
-    UCS_INIT_ONCE(&init_once) {
-        size = ucs_read_file_str(guid_str, sizeof(guid_str), 1,
-                                 "%s", boot_id_filename);
-        if (size <= 0) {
-            goto use_machine_guid;
-        }
-
-        res = sscanf(guid_str, guid_scan_fmt,
-                     ucs_unaligned_ptr(&guid.v1),
-                     ucs_unaligned_ptr(&guid.v2),
-                     ucs_unaligned_ptr(&guid.v3),
-                     ucs_unaligned_ptr(&guid.v4),
-                     &guid.v5[0], &guid.v5[1], &guid.v5[2],
-                     &guid.v5[3], &guid.v5[4], &guid.v5[5]);
-        if (res != 10) { /* 10 values should be scanned */
-            goto use_machine_guid;
-        }
-
-        system_id = guid.hi_dword ^ guid.low_dword;
-        continue;
-
-use_machine_guid:
-        system_id = ucs_machine_guid();
+    uint64_t high;
+    uint64_t low;
+    ucs_status_t status;
+
+    status = ucs_sys_get_boot_id(&high, &low);
+    if (status == UCS_OK) {
+        return high ^ low;
     }
 
-    return system_id;
+    return ucs_machine_guid();
 }
 
 ucs_status_t UCS_F_NOOPTIMIZE /* GCC failed to compile it in release mode */
 uct_sm_iface_get_device_address(uct_iface_t *tl_iface, uct_device_addr_t *addr)
 {
     ucs_sm_iface_ext_device_addr_t *ext_addr = (void*)addr;
 
-    ext_addr->super.id  = uct_sm_iface_get_system_id();
-    ext_addr->super.ext = !ucs_sys_ns_is_root(UCS_SYS_NS_IPC);
+    ext_addr->super.id  = uct_sm_iface_get_system_id() & ~UCS_SM_IFACE_ADDR_FLAG_EXT;
+    ext_addr->super.id |= ((uint64_t)!ucs_sys_ns_is_default(UCS_SYS_NS_TYPE_IPC)) << 63;
```

Comment:
reuse UCS_SM_IFACE_ADDR_FLAG_EXT instead of '63'

Reply:
ok, reused